src/hooks/useTrackRegistration.ts

```diff
@@ -25,14 +34,23 @@ export function useTrackRegistration() {
       return;
     }
 
-    // Skip if the sign-in has already been tracked
+    // Skip if the sign-in has already been tracked on a previous page
     if (
       user.unsafeMetadata.owa?.lastTrackedSignInAt ===
       user.lastSignInAt?.valueOf()
     ) {
       return;
     }
 
+    // Skip tracking when the hook has been re-mounted/re-rendered
+    // this guards against a race condition caused by the hook being unmounted
+    // and re-rendered while the Clerk user metadata is being updated
+    if (hasAlreadyTrackedEvent) {
```

Comment:
I was relying on the dependencies array to prevent the hook from firing again on subsequent renders while the user metadata update was taking place. I really should have known better than to trust `useEffect` in this scenario.

It took me a while to get to the bottom of what was going on but ultimately the hook is unmounted repeatedly.
1. Once on initial page load (when the Bugsnag error boundary is added to the tree)
2. On every navigation through the pages router (I didn't know this about the pages router, this is different in app router IIRC where the tree isn't thrown away).

It was on navigation that things were going wrong:

1. The user completes sign-up
3. We redirect to the lesson download page
4. The hook fires a few times
5. Dependencies settle when the user is loaded through `useUser`
    * The hook fires with all dependencies met
    * The tracking call is made
    * A request is made to update user metadata in Clerk with the last tracked sign-in (this is not an optimistic update)
6. In or around the same tick that the user is loaded we redirect to `/onboarding`
   * The hook is remounted throwing away all state
   * The hook fires immediately since all dependencies have been met
     * Critically the update user metadata request has not responded yet
   * The tracking call is made again
   * Another request is sent to update user metadata with the last tracked sign-in
 7. Both requests to update user metadata eventually respond and the last tracked date sign in settles.